recipes/cppcheck/all/conanfile.py

```diff
@@ -27,20 +34,33 @@ def _build_subfolder(self):
         return "build_subfolder"
 
     def _patch_sources(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
-        tools.replace_in_file(os.path.join(self._source_subfolder, "cli", "CMakeLists.txt"),
+        apply_conandata_patches(self)
```

Comment:
This patch is annoying to maintain. I think using `replace_in_file` is more flexible.